Gems/Atom/RPI/Code/Source/RPI.Private/RPISystemComponent.cpp

```diff
@@ -185,21 +187,36 @@ namespace AZ
         AZ_CVAR_EXTERNED(AZ::CVarFixedString, r_metricsDataLogType);
         AZ_CVAR_EXTERNED(AZ::u32, r_metricsWaitTimePerCaptureBatch);
         AZ_CVAR_EXTERNED(AZ::u32, r_metricsFrameCountPerCaptureBatch);
+        AZ_CVAR_EXTERNED(bool, r_metricsQuitUponCompletion);
+
+        AZStd::string RPISystemComponent::GetLogCategory()
+        {
+            AZStd::string platformName = AZ::GetPlatformName(AZ::g_currentPlatform);
+            AZ::Name apiName = AZ::RHI::Factory::Get().GetName();
+            auto logCategory = AZStd::string::format("%.*s-%s-%s", AZ_STRING_ARG(PerformanceLogCategory), platformName.c_str(), apiName.GetCStr());
+            return logCategory;
+        }
 
         void RPISystemComponent::InitializePerformanceCollector()
         {
             auto onBatchCompleteCallback = [](AZ::u32 pendingBatches) {
                 AZ_TracePrintf("RPISystem", "Completed a performance batch, still %u batches are pending.\n", pendingBatches);
                 r_metricsNumberOfCaptureBatches = pendingBatches;
+                if (r_metricsQuitUponCompletion && (pendingBatches == 0))
+                {
+                    AzFramework::ConsoleRequestBus::Broadcast(
+                        &AzFramework::ConsoleRequests::ExecuteConsoleCommand, "quit");
+                }
             };
 
             auto performanceMetrics = AZStd::to_array<AZStd::string_view>({
                 PerformanceSpecGraphicsSimulationTime,
                 PerformanceSpecGraphicsRenderTime,
                 PerformanceSpecEngineCpuTime,
                 });
+            AZStd::string logCategory = GetLogCategory();
```

Comment:
nit: no need to store in a local variable, function can be called as part of the make_unique call.

Reply:
Yes, but I do that to facilitate debugging.